src/mtconnect/buffer/circular_buffer.hpp

```diff
@@ -82,11 +85,29 @@ namespace mtconnect::buffer {
     /// @param diMap the map of data item ids to new data item entities
     void updateDataItems(std::unordered_map<std::string, WeakDataItemPtr> &diMap)
     {
-      for (auto &o : m_slidingBuffer)
+      std::vector<boost::circular_buffer<observation::ObservationPtr>::iterator> orphanBufferItems;
+
+      auto iter = m_slidingBuffer.begin();
+      while( iter != m_slidingBuffer.end() )
       {
-        o->updateDataItem(diMap);
+        observation::ObservationPtr o = *iter;
+        if( o->isOrphan() ) {
+          orphanBufferItems.push_back(iter);
+        }
+        else {
+          o->updateDataItem(diMap);
+        }
+        iter++;
+      }
+
+      // remove orphans from the slidingBuffer
+      while(!orphanBufferItems.empty()) {
+        auto orphanIterIter = orphanBufferItems.back();
+        m_slidingBuffer.erase(orphanIterIter);
```

Comment:
We can't erase from the sliding buffer. The issue is it messes up the sequence number computation. I'll check into this and see if I can work around the issue. It should be sufficient to have them orphaned. Did you try without this code?

Reply:
It appears to work without erasing the orphaned dataItems from the sliding buffer. I'm testing my changes and will commit them soon.